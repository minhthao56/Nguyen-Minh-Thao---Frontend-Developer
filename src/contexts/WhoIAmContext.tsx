import { createContext, useContext } from "react";
import { AuthResp } from "../services/user";

export interface WhoIAmContextReturn extends AuthResp {
  isLoading?: boolean;
}

const WhoIAmContextContext = createContext<WhoIAmContextReturn>({
  email: "",
  token: "",
  isLoading: false,
});
WhoIAmContextContext.displayName = "WhoIAmContextContext";
const useWhoIAmContext = () => useContext(WhoIAmContextContext);
export { WhoIAmContextContext, useWhoIAmContext };
