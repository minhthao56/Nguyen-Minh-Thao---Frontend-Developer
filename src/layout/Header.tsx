import Logo from "../assets/img/logo.png";

export default function Header() {
  return (
    <div className="relative bg-black">
      <div className="mx-auto max-w-7xl px-6">
        <div className="flex items-center justify-between py-6 md:justify-start md:space-x-10">
          <div className="flex justify-start lg:w-0 lg:flex-1">
            <a href="/">
              <span className="sr-only">Your Company</span>
              <img className="h-8 w-auto sm:h-10" src={Logo} alt="logo" />
            </a>
          </div>
          <nav className="hidden space-x-10 md:flex">
            <a
              href="/"
              className="text-base font-medium text-white hover:text-gray-900"
            >
              Pricing
            </a>
            <a
              href="/"
              className="text-base font-medium text-white hover:text-gray-900"
            >
              Docs
            </a>
          </nav>
          <div className="hidden items-center justify-end md:flex md:flex-1 lg:w-0">
            <a
              href="/"
              className="whitespace-nowrap text-base font-medium text-white"
            >
              Sign in
            </a>
            <a
              href="/"
              className="ml-8 inline-flex items-center justify-center whitespace-nowrap rounded-md border border-transparent bg-white px-4 py-2 text-base font-medium text-black shadow-sm"
            >
              Sign up
            </a>
          </div>
        </div>
      </div>
    </div>
  );
}
