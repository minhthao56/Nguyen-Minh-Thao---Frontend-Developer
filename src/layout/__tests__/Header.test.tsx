import { render, screen } from "@testing-library/react";
import Header from "../Header";
import { useWhoIAmContext } from "../../contexts/WhoIAmContext";
import { BrowserRouter } from "react-router-dom";

jest.mock("../../contexts/WhoIAmContext", () => {
  return {
    useWhoIAmContext: jest.fn(),
  };
});

describe("<Header/>", () => {
  beforeEach(() => {
    (useWhoIAmContext as jest.Mock).mockReturnValue({ email: "123@gmail.com" });
  });

  const renderComponent = () => {
    return render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
  };

  it("should match to snapshot when having auth", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(screen.getByText("123@gmail.com")).toBeInTheDocument();
  });

  it("should match to snapshot when non-auth", () => {
    (useWhoIAmContext as jest.Mock).mockReturnValue({ email: "" });
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(screen.queryByText("123@gmail.com")).not.toBeInTheDocument();

    expect(screen.getByRole("button", { name: "Login" })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: "Sign in" })).toBeInTheDocument();
  });
});
