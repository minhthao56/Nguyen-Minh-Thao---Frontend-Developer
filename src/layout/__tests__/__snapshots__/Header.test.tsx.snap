// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match to snapshot when having auth 1`] = `
<div>
  <div
    class="relative bg-black"
  >
    <div
      class="mx-auto max-w-7xl px-6"
    >
      <div
        class="flex items-center justify-between py-6 md:justify-start md:space-x-10"
      >
        <div
          class="flex justify-start lg:w-0 lg:flex-1"
        >
          <a
            href="/"
          >
            <span
              class="sr-only"
            >
              Your Company
            </span>
            <img
              alt="logo"
              class="h-8 w-auto sm:h-10"
              src="logo.png"
            />
          </a>
        </div>
        <div
          class="hidden items-center justify-end md:flex md:flex-1 lg:w-0"
        >
          <span
            class="text-white"
          >
            123@gmail.com
          </span>
          <button
            class="text-white bg-black focus:ring-4 focus:outline-none focus:ring-black font-medium rounded-lg text-sm w-full px-5 py-2.5 text-center w-14 ml-8"
            type="button"
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Header/> should match to snapshot when non-auth 1`] = `
<div>
  <div
    class="relative bg-black"
  >
    <div
      class="mx-auto max-w-7xl px-6"
    >
      <div
        class="flex items-center justify-between py-6 md:justify-start md:space-x-10"
      >
        <div
          class="flex justify-start lg:w-0 lg:flex-1"
        >
          <a
            href="/"
          >
            <span
              class="sr-only"
            >
              Your Company
            </span>
            <img
              alt="logo"
              class="h-8 w-auto sm:h-10"
              src="logo.png"
            />
          </a>
        </div>
        <div
          class="hidden items-center justify-end md:flex md:flex-1 lg:w-0"
        >
          <a
            href="/register"
          >
            <button
              class="text-white bg-black focus:ring-4 focus:outline-none focus:ring-black font-medium rounded-lg text-sm w-full px-5 py-2.5 text-center w-40"
              type="button"
            >
              Sign in
            </button>
          </a>
          <a
            href="/login"
          >
            <button
              class="text-white bg-black focus:ring-4 focus:outline-none focus:ring-black font-medium rounded-lg text-sm w-full px-5 py-2.5 text-center w-40"
              type="button"
            >
              Login
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
